### DISCUSSION TITLE
Storing computations of vertex shader in buffer

### DISCUSSION THREAD

**Discussion by noergel1** [google/filament#6698] [Q&A] [answered]
Hey,

is it possible to store computations done in the vertex shader persistently, so that they don't have to be redone every frame?
I want to implement triplanar mapping, so I'd like store the calculated UVs in a buffer.

Thanks for your help!

**Comment by bejado** [comment, marked as answer]
Yes, you can store any computation you want inside vertex attributes. We provide 8 attributes for your own use: custom0-7. Then, in the material you can mark them as "required" and access them via `getCustom0()`. https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/vertexandattributes:requires

Note that these attributes must be computed on the CPU.

**Reply by noergel1** [reply]
> Yes, you can store any computation you want inside vertex attributes. We provide 8 attributes for your own use: custom0-7. Then, in the material you can mark them as "required" and access them via `getCustom0()`. https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/vertexandattributes:requires
> 
> Note that these attributes must be computed on the CPU.

Alright, thank you. I actually hoped there'd be a way to store GPU computations in a buffer, do you happen to know if thats planned to be implemented?

**Reply by bejado** [reply]
We're eventually adding support for SSBOs (we already have some support for them internally), but this work isn't completed yet and we don't have a definite timeline.
